### PR TITLE
fix IPv6 static default gateway not set if gateway is a LL unicast address

### PR DESCRIPTION
**Pull request by freswa**
# Pull Request
## What? (description)
Fix IPv6 default gateway creation when the gateway is only reachable via Link-Local-Unicast address.

## Why? (reasoning)
It was broken. Discussed on Slack with @andrewrynhard 

## Acceptance

Please use the following checklist:

- [ ] you linked an issue (if applicable)
- [ ] you included tests (if applicable)
- [x] you ran conformance (`make conformance`)
- [x] you formatted your code (`make fmt`)
- [x] you linted your code (`make lint`)
- [x] you generated documentation (`make docs`)
- [x] you ran unit-tests (`make unit-tests`)

> See `make help` for a description of the available targets.

---
This change is [Reviewable](https://reviewable.io/reviews/talos-systems/talos/2075)

### CODE REVIEW

**Comment by andrewrynhard** [comment]
@freswa Looks like we just need the commit signed with the DCO (`git commit -s --amend --no-edit`)

**Comment by freswa** [comment]
Sorry for all the force-pushes. Should have looked at the checks before I create a PR :(

**Comment by andrewrynhard** [comment]
No worries at all. We force push all the time

File: internal/app/networkd/pkg/nic/nic.go [hunk 1]
```diff
@@ -366,10 +366,15 @@ func (n *NetworkInterface) configureInterface(method address.Addressing, link *n
 		if net.IPv4zero.Equal(gw) {
 			gw = nil
 		}
+		src := method.Address()
```
Comment (andrewrynhard) [inline internal/app/networkd/pkg/nic/nic.go:369]: Linter will hit you here. Need a line above this.

**Pull request merged**